tests/multipleBackend/multipartUpload.js

```diff
@@ -0,0 +1,169 @@
+                    s3.abortMultipartUpload(awsParams, err => {
+                        assert.strictEqual(err, null,
+                            `Error aborting MPU ${err}`);
+                        done();
+                    });
+                });
+            });
+        });
+    });
+
+    it('should complete a multipart upload on real AWS', done => {
+        async.waterfall([
+            next => bucketPut(authInfo, bucketPutRequest, log, err =>
+                next(err)
+            ),
+            next => initiateMultipartUpload(authInfo, initiateRequest,
+            log, (err, result) =>
+                next(err, result)
+            ),
+            (result, next) => {
+                parseString(result, (err, json) => {
+                    const uploadId =
+                        json.InitiateMultipartUploadResult.UploadId[0];
+                    return next(err, uploadId);
+                });
+            },
+            (uploadId, next) => {
+                partParams.url =
+                    `/${objectKey}?partNumber=1&uploadId=${uploadId}`;
+                partParams.query = { partNumber: '1', uploadId };
+                const partRequest = new DummyRequest(partParams, postBody);
+                objectPutPart(authInfo, partRequest, undefined, log,
+                err =>
+                    next(err, uploadId)
+                );
+            },
+            (uploadId, next) => {
+                awsParams.UploadId = uploadId;
+                s3.listParts(awsParams, (err, data) => {
+                    assert.strictEqual(data.Parts.length, 1);
+                    return next(err, uploadId);
+                });
+            },
+            (uploadId, next) => {
+                completeParams.url = `/${objectKey}?uploadId=${uploadId}`;
+                completeParams.query = { uploadId };
+                completeMultipartUpload(authInfo, completeParams, log,
+                (err, result) =>
+                    next(err, result)
+                );
+            },
+            (result, next) => {
+                parseString(result, (err, json) => {
+                    assert.strictEqual(
+                        json.CompleteMultipartUploadResult.Location[0],
+                        `http://${bucketName}.s3.amazonaws.com`
+                        + `/${objectKey}`);
+                    assert.strictEqual(json.CompleteMultipartUploadResult
+                        .Bucket[0], bucketName);
+                    assert.strictEqual(
+                        json.CompleteMultipartUploadResult.Key[0],
+                        objectKey);
+                    const MD = metadata.keyMaps.get(bucketName).get(objectKey);
+                    assert(MD);
+                    return next(err);
+                });
+            },
+        ], err => {
+            assert.strictEqual(err, null, `Error completing MPU: ${err}`);
+            done();
+        });
+    });
+});
```

Comment:
we should also be testing listing of mpu's and separately testing listing of parts.